jina/serve/runtimes/gateway/grpc/__init__.py

```diff
@@ -127,3 +130,34 @@ async def async_run_forever(self):
         """The async running of server."""
         self._connection_pool.start()
         await self.server.wait_for_termination()
+
+    async def dry_run(self, empty, context) -> jina_pb2.StatusProto:
+        """
+        Process the the call requested by having a dry run call to every Executor in the graph
+
+        :param empty: The service expects an empty protobuf message
+        :param context: grpc context
+        :returns: the response request
+        """
+        from docarray import DocumentArray
+        from jina.clients.request import request_generator
+        from jina.enums import DataInputType
+        from jina.serve.executors import __dry_run_endpoint__
+
+        da = DocumentArray()
+
+        try:
+            req_iterator = request_generator(
+                exec_endpoint=__dry_run_endpoint__,
+                data=da,
+                data_type=DataInputType.DOCUMENT,
+            )
+            async for _ in self.streamer.stream(request_iterator=req_iterator):
+                pass
```

Comment:
Same as above maybe log something here. The goal would be that if someone want to understand where something is failing he can check from the log that the health check as been send from the gateway but not receive in the Executor